### PR TITLE
Fix empty array handling in arraySplit

### LINKED ISSUE #7708: arraySplit usage lead to: Data compressed with different methods exception
```
Exception on client:
Code: 271. DB::Exception: Data compressed with different methods, given method byte 00, previous method byte 82: while receiving packet from localhost:9000
```

### PR DESCRIPTION
**Pull request by hczhcz**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (up to few sentences, required except for Non-significant/Documentation categories):

Fix #7708 empty array handling in `arraySplit`

### CODE REVIEW

**Review by tavplubix — COMMENTED** [review]
In case like `SELECT arraySplit(x -> 0, [])` column `mapped` is `ColumnConst` and `mapped->size()` is zero. However we access it's value in `if (column_cut_const->getValue<UInt8>())` (line 68). It works because nested column actually has size 1 and `ColumnConst` just returns value of nested column without any checks, but it looks very suspicious.

**Comment by hczhcz** [comment]
@tavplubix 
`...->getValue<UInt8>()` is used in many high-order functions (`arrayFilter`, `arrayCount`, `arrayAll`, etc).
It might be good to make developers aware the suspicious thing by adding some comments & docs.

File: dbms/tests/queries/0_stateless/01015_array_split.reference [hunk 1]
```diff
@@ -6,8 +6,10 @@
 [[1],[2],[3],[4],[5]]
 [[1,2],[3,4],[5]]
 [[1],[2,3],[4,5]]
-[[]]
-[[]]
+[]
```
[the remark this answers is not in the document]
Comment (alexey-milovidov) [reply dbms/tests/queries/0_stateless/01015_array_split.reference]: Ok for now.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by nikitamikhaylov** [comment]
Conflicts with 19.16

**Pull request merged**